src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -384,17 +447,92 @@ ImmutableArray<IMemberInitializer> IObjectCreationExpression.MemberInitializers
         {
             get
             {
-                BoundExpression initializer = this.InitializerExpressionOpt;
-                if (initializer != null)
-                {
-                    // ZZZ What's the representation in bound trees?
-                }
-
-                return ImmutableArray.Create<IMemberInitializer>();
+                var initializers = s_memberInitializersMappings.GetValue(this,
+                    objCreationExp =>
+                    {
+                        BoundExpression initializer = this.InitializerExpressionOpt;
+                        if (initializer != null)
+                        {
+                            var objInitializerExp = initializer as BoundObjectInitializerExpression;
+                            if (objInitializerExp != null)
+                            {
+                                var builder = ArrayBuilder<IMemberInitializer>.GetInstance(objInitializerExp.Initializers.Length);
+                                foreach (var memberAssignment in objInitializerExp.Initializers)
+                                {
+                                    var assignment = memberAssignment as BoundAssignmentOperator;
+                                    var leftSymbol = (assignment?.Left as BoundObjectInitializerMember)?.MemberSymbol;
+
+                                    if (leftSymbol == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    switch (leftSymbol.Kind)
+                                    {
+                                        case SymbolKind.Field:
+                                            builder.Add(new FieldInitializer(assignment.Syntax, (IFieldSymbol)leftSymbol, assignment.Right));
+                                            break;
+                                        case SymbolKind.Property:
+                                            builder.Add(new PropertyInitializer(assignment.Syntax, ((IPropertySymbol)leftSymbol).SetMethod, assignment.Right));
+                                            break;
+                                    }
+                                }
+                                return builder.ToArrayAndFree();
+                            }
+                        }
+                        return new IMemberInitializer[] { };
+                    });
+                var immBuilder = ArrayBuilder<IMemberInitializer>.GetInstance(initializers.Length);
+                immBuilder.AddRange(initializers);
+                return immBuilder.ToImmutableAndFree();                
```

Comment:
As discussed offline, I'm boxing the array to avoid creating a new one for every request.